### DISCUSSION TITLE
Empty array of a very specific type?

### DISCUSSION THREAD

**Discussion by diggyk** [diwic/dbus-rs#361] [Q&A] [answered]
I'm trying to send a `StartTransientUnit` to the SystemD manager, but it has a very peculiar part of the spec.  For the last argument, it expects an `a(sa(sv))` but the specs say this is unused and you should just send `0` here for the value.  

So when I look at a valid message using the bus monitor, I see it looks like this:
```
          ARRAY "(sa(sv))" {
          };
```

But I can't figure out how to craft that using this crate.  Any pointers?

**Comment by diwic** [comment, marked as answer]
How about:

    let x: Vec<(String, Vec<(String, Variant)>)> = vec!();

...then you can do whatever you want to do with `x`, e g use it in `message.append1(x)` or `Message::call_with_args`.

**Reply by diggyk** [reply]
Perfect!  That worked beautifully!  Thank you so much.  Love this crate so much!
